shell/browser/api/electron_api_web_contents.h

```diff
@@ -803,6 +808,8 @@ class WebContents : public ExclusiveAccessContext,
 
   scoped_refptr<base::SequencedTaskRunner> file_task_runner_;
 
+  std::queue<base::ScopedClosureRunner> capture_handles_;
```

Comment:
I _think_ this scheme works iff the promises are guaranteed to resolve in the same order as this queue. Is this guaranteed to be true?

(Even if that's not guaranteed, a modified version of this might work, e.g. some kind of mapping to look up the right ScopedClosureRunner)